### ISSUE TITLE
Does grpc still not support string array

### ISSUE THREAD

**Issue by godcong** [ent/ent#1585]
I used ent to create my grpc interface, but the following error occurred:
```
entproto: failed generating protos: entproto: failed parsing some schemas: unsupported field type "TypeJSON"
```

**Comment by rotemtam** [comment]
True, you can find the support and mapping matrix on the package readme.

**Comment by rotemtam** [comment]
Having said that, if you want to contribute support for this use case, I will be happy to help

**Comment by godcong** [comment]
I feel that I may not be able to deal with this temporarily.
I haven't understood some implementations of `ent gen`.

**Comment by rotemtam** [comment]
@godcong it should not be a very difficult implementation. if you like ping me on slack and I will break it down with you into small steps. I will b very happy to have more people contributing to this part of ent

**Comment by jordanrule** [comment]
I got this generating protos as a `google.protobuf.ListValue`, but of course the type of the internal type is lost in the proto.  I think the cleanest way to do this is to add a `repeatedType` field in the `typeConfig` struct and treat it as a special case in the adapter (like `TypeEnum`).  Coming back to this as I have time but I feel like I almost have it, as I become more comfortable with the adapter code.

**Comment by rotemtam** [comment]
For the special case of field.Strings I think (as mentioned above) it should be mapped to a repeated string field. Don't think ListValue is needed

**Comment by jordanrule** [comment]
I'm not sure I understand, there is no type for repeated in the descriptor (https://github.com/protocolbuffers/protobuf-go/blob/fb30439f551a7e79e413e7b4f5f4dfb58e117d73/types/descriptorpb/descriptor.pb.go#L53), the descriptor notes ListValue is the type (https://github.com/protocolbuffers/protobuf-go/blob/fb30439f551a7e79e413e7b4f5f4dfb58e117d73/types/known/structpb/struct.pb.go#L554) which I mapped to a slice of interface as a primitive in my first version.  Like I said it came out messy in the proto, which is why I am treating it as a special case to arrive at your specification.

**Comment by rotemtam** [comment]
It's not a type, it's called a label in the proto descriptor lingo. The type would be string and the field would have a label of repeated

**Comment by jordanrule** [comment]
Okay, I did see the repeated label but I wasn't sure it would work in tandem with the string type - I will use that approach.  Thank you for being patient and explicit while I learn!

**Comment by rotemtam** [comment]
Sure thing, thanks for working on this!

**Comment by godcong** [comment]
@rotemtam 
this issue now looks okay, can closed?
